creator-node/src/dbManager.js

```diff
@@ -180,24 +180,50 @@ class DBManager {
     prevStoragePath,
     batchSize
   ) {
-    const files = (
+    // Get a batch of file that this user needs
+    const userFiles = (
       await models.File.findAll({
         attributes: ['storagePath'],
         where: {
-          [sequelize.Op.and]: [
-            sequelize.literal(`"multihash" IN
-          (SELECT "multihash" FROM "Files" WHERE "cnodeUserUUID" = '${cnodeUserUUID}' AND "storagePath" > '${prevStoragePath}')`)
-          ]
+          cnodeUserUUID,
+          storagePath: {
+            [sequelize.Op.gte]: prevStoragePath
+          }
         },
-        group: 'storagePath',
-        having: sequelize.literal(`COUNT(DISTINCT("cnodeUserUUID")) = 1`),
+        distinct: true,
         order: [['storagePath', 'ASC']],
         limit: batchSize
       })
-    ).map((result) => result.dataValues)
+    )
+      .map((result) => result.dataValues)
```

Comment:
since we're dealing with larger amounts of data can we do both these things in one loop vs two maps?

Reply:
changed